bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/AutoUpdateManager.java

```diff
@@ -174,27 +176,41 @@ public void receiveCommand(ItemCommandEvent commandEvent, Item item) {
     private Recommendation shouldAutoUpdate(String itemName) {
         Recommendation ret = Recommendation.REQUIRED;
 
-        Set<ChannelUID> linkedChannelUIDs = itemChannelLinkRegistry.stream().filter(link -> {
-            return link.getItemName().equals(itemName);
-        }).map(link -> {
-            return link.getLinkedUID();
-        }).collect(toSet());
+        List<ChannelUID> linkedChannelUIDs = new ArrayList<>();
+        for (ItemChannelLink link : itemChannelLinkRegistry.getAll()) {
+            if (link.getItemName().equals(itemName)) {
+                linkedChannelUIDs.add(link.getLinkedUID());
+            }
+        }
+
+        // check if there is any channel ONLINE
+        List<ChannelUID> onlineChannelUIDs = new ArrayList<>();
         for (ChannelUID channelUID : linkedChannelUIDs) {
             Thing thing = thingRegistry.get(channelUID.getThingUID());
             if (thing == null //
                     || thing.getChannel(channelUID.getId()) == null //
                     || thing.getHandler() == null //
                     || !ThingStatus.ONLINE.equals(thing.getStatus()) //
             ) {
-                if (ret == Recommendation.REQUIRED) {
-                    ret = Recommendation.REVERT;
-                }
                 continue;
             }
+            onlineChannelUIDs.add(channelUID);
+        }
+        if (!linkedChannelUIDs.isEmpty() && onlineChannelUIDs.isEmpty()) {
+            // none of the linked channels is able to process the command
+            return Recommendation.REVERT;
+        }
 
-            AutoUpdatePolicy policy = thing.getChannel(channelUID.getId()).getAutoUpdatePolicy();
-            if (policy == null) {
-                policy = AutoUpdatePolicy.DEFAULT;
+        for (ChannelUID channelUID : onlineChannelUIDs) {
+            Thing thing = thingRegistry.get(channelUID.getThingUID());
+            if (thing == null) {
+                // this should not happen, but we make avoid null warnings from the compiler
```

Comment:
But I assume the thing can be null if it has been removed from the registry (provider) while this method is executed (e.g. deleted by user request).

Reply:
You are right. I have removed the comment with https://github.com/eclipse/smarthome/pull/6194 now.